components/places/src/storage.rs

```diff
@@ -161,11 +171,17 @@ pub fn apply_observation_direct(db: &Connection, visit_ob: VisitObservation) ->
             if !visit_ob.is_error.unwrap_or(false) {
                 update_frecency = true;
             }
+            update_change_counter = true;
             Some(row_id)
         },
         None => None,
     };
 
+    if update_change_counter {
+        page_info.sync_change_counter += 1;
```

Comment:
I think this is wrong - iiuc sqlite's concurrency, the sql should read `sync_change_counter = sync_change_counter + 1` instead? (or in practice, `sync_change_counter = sync_change_counter + :update_change_counter`)

Reply:
Don't we have a transaction out now?

Reply:
My preference is to do this in SQL, too, but, as @thomcc says, if we have a transaction active, it shouldn't really matter.